Python/pylifecycle.c

```diff
@@ -760,6 +760,11 @@ pycore_init_types(PyInterpreterState *interp)
         return status;
     }
 
+    status = _PyDateTime_InitTypes(interp);
+    if (_PyStatus_EXCEPTION(status)) {
+        return status;
+    }
```

Comment:
```suggestion
    if (!_Py_IsMainInterpreter(interp)) {
        status = _PyDateTime_InitTypes(interp);
        if (_PyStatus_EXCEPTION(status)) {
            return status;
        }
    }
```

Reply from https://github.com/python/cpython/pull/136620#issuecomment-3067427528

Can you run `test_concurrent_initialization()` with this change?  Based on the crashes that come from the change, I said this PR and my example are "almost equivalent."  I'm not sure right now what this PR ensures.

Reply:
Hm, what are you trying to achieve here? This will just break the types for the main interpreter.

Reply:
> This will just break the types for the main interpreter.

Note that `test_concurrent_initialization()` does not load the `_datetime` in the main inter interpreter at all.

Reply:
> Can you run test_concurrent_initialization() with this change?

Correction: Run the *script* of the test.